src/Engine/Spells/CastSpellInfo.cpp

```diff
@@ -829,20 +817,20 @@ void CastSpellInfoHelpers::castSpell() {
                         pParty->pPlayers[pCastSpell->uPlayerID_2].pPlayerBuffs[PLAYER_BUFF_BLESS]
                             .Apply(pParty->GetPlayingTime() + spell_duration, spell_mastery, spell_power, spell_overlay_id, 0);
                     } else {
-                        for (uint pl_id = 0; pl_id < 4; pl_id++) {
-                            spell_fx_renderer->SetPlayerBuffAnim(pCastSpell->uSpellID, pl_id);
-                            int spell_overlay_id = pOtherOverlayList->_4418B1(10000, pl_id + 310, 0, 65536);
-                            pParty->pPlayers[pl_id].pPlayerBuffs[PLAYER_BUFF_BLESS]
+                        for (size_t i = 0; Player &PlayerRef : pParty->pPlayers) {
```

Comment:
Codestyle for `PlayerRef`. Also I'm not a fan of loops like this one, as the explicit for loop with indices is more readable in this case. But this one's up to you.

Reply:
Should be doable with ranges-v3, smth like (pseudocode)
```
for(auto && [i, player] : pParty->pPlayers | ranges::enumerate) { ...}
```

But not really worth the effort.